core/src/cluster_info.rs

```diff
@@ -1437,18 +1437,26 @@ impl ClusterInfo {
             .iter()
             .enumerate()
             .map(|(i, (responses, _from_addr))| {
-                let score: u64 = if stakes.get(&responses[0].pubkey()).is_some() {
+                let stake_score: u64 = if stakes.get(&responses[0].pubkey()).is_some() {
                     2
                 } else {
                     1
                 };
                 responses
                     .iter()
                     .enumerate()
-                    .map(|(j, _response)| ResponseScore {
-                        to: i,
-                        responses_index: j,
-                        score,
+                    .map(|(j, response)| {
+                        let response_score = match response.data {
+                            CrdsData::ContactInfo(_) => 3,
```

Comment:
hmmm instead of a score, which is still probabilistic do you think it would be more effective to:

1) Have a quota per validator
2) For each validator, sort their responses to prioritize ContactInfo, Vote, AccountHashes, etc.

That way each validator gets a guaranteed quota and the prioritization of important messages is deterministic

Reply:
It's a lot more work computationally